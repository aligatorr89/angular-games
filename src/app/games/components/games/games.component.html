<app-games-header (changeNameEmmiter)="getGames($event)" (changeCategoryEmmiter)="getGames($event)"></app-games-header>
<app-games-list *ngIf="games$ | async as games" [games]="games"></app-games-list>
