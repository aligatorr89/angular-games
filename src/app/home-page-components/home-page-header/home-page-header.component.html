<h1 class="text-center">
  Welcome to Angular Games!
</h1>
