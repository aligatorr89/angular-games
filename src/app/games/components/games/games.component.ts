import { Component, OnInit } from '@angular/core';
import { GamesService } from '../../services/games.service';
import { Observable } from 'rxjs';
import { Game, GameQuery } from '../../../shared/models/game';

@Component({
  selector: 'app-games',
  templateUrl: './games.component.html',
  styleUrls: ['./games.component.css']
})
export class GamesComponent implements OnInit {

  games$: Observable<Game[]>;

  constructor(private gamesService: GamesService) { }

  ngOnInit() {
    this.getGames();
  }

  getGames(query?: GameQuery) {
    this.games$ = this.gamesService.getGames(query);
  }

}
