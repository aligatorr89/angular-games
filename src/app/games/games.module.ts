import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { GamesComponent } from './components/games/games.component';
import { GamesListComponent } from './components/games-list/games-list.component';
import { GamesListElementComponent } from './components/games-list-element/games-list-element.component';
import { GamesHeaderComponent } from './components/games-header/games-header.component';

@NgModule({
  declarations: [GamesComponent, GamesListComponent, GamesListElementComponent, GamesHeaderComponent],
  imports: [
    CommonModule, FormsModule
  ],
  exports: [
    GamesComponent
  ]
})
export class GamesModule { }
