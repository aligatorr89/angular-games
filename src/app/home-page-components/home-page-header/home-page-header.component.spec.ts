import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { HomePageHeaderComponent } from './home-page-header.component';
import { GamesModule } from '../../games/games.module';
import { ComponentTestInit } from '../../shared/test/test-util';

describe('HomePageHeaderComponent', () => {
  let test: ComponentTestInit<HomePageHeaderComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ HomePageHeaderComponent ],
      imports: [ GamesModule ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    test = new ComponentTestInit(HomePageHeaderComponent);
  });

  it('should create', () => {
    expect(test.component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(test.html.outerHTML).toMatchSnapshot();
  });

});
