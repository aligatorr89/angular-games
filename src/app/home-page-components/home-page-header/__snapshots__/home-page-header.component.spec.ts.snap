// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageHeaderComponent should match snapshot 1`] = `"<div id=\\"root1\\" ng-version=\\"8.0.3\\"><nav class=\\"navbar navbar-expand-lg navbar-light bg-light\\"><a class=\\"navbar-brand\\" routerlink=\\"\\">Angular Games</a><button aria-controls=\\"navbarSupportedContent\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\" class=\\"navbar-toggler\\" data-target=\\"#navbarSupportedContent\\" data-toggle=\\"collapse\\" type=\\"button\\"><span class=\\"navbar-toggler-icon\\"></span></button><div class=\\"collapse navbar-collapse\\" id=\\"navbarSupportedContent\\"><ul class=\\"navbar-nav mr-auto\\"><li class=\\"nav-item active\\"><a class=\\"nav-link\\" routerlink=\\"\\">Home<span class=\\"sr-only\\">(current)</span></a></li><li class=\\"nav-item active\\"><a class=\\"nav-link\\" routerlink=\\"/games\\">Games</a></li></ul><form class=\\"form-inline my-2 my-lg-0\\"><input aria-label=\\"Username\\" class=\\"form-control mr-sm-2\\" placeholder=\\"Username\\" type=\\"search\\"><input aria-label=\\"Password\\" class=\\"form-control mr-sm-2\\" placeholder=\\"Password\\" type=\\"password\\"><button class=\\"btn btn-outline-success my-2 my-sm-0\\" type=\\"submit\\">Login</button></form></div></nav></div>"`;
